### ISSUE TITLE
Eval bug:

### ISSUE THREAD

**Issue by evangelinehelsinki** [ggml-org/llama.cpp#20659]
### Name and Version

$ ./llama-server --version
version: 8363 (ceef6b523)
built with GNU 13.3.0 for Linux x86_64

### Operating systems

Linux

### GGML backends

CUDA

### Hardware

RTX 3090 24GB (Ampere, SM 8.6)

### Models

GLM-4.7-Flash (deepseek2 architecture), Q4_K_M quantization.
tested three separate GGUF providers to rule out conversion issues:
- https://huggingface.co/bartowski/zai-org_GLM-4.7-Flash-GGUF (Q4_K_M)
- https://huggingface.co/unsloth/GLM-4.7-Flash-GGUF (Q4_K_M)
- https://huggingface.co/ngxson/GLM-4.7-Flash-GGUF (Q4_K_M)

### Problem description & steps to reproduce

GLM-4.7-Flash produces a static logit distribution on CUDA. he same output tokens regardless of input or position. CPU inference on the same GGUF returns correct results. the key diagnostic seems to be that with `temperature=0` and `logprobs=true`, every single position returns the same top-3 tokens:
```
pos 1: selected=id:30 '?'  top: [(103257,'有时候'), (103248,'成为了'), (103249,'武器')]
pos 2: selected=id:30 '?'  top: [(103257,'有时候'), (103248,'成为了'), (103249,'武器')]
pos 3: selected=id:30 '?'  top: [(103257,'有时候'), (103248,'成为了'), (103249,'武器')]
```
nothing about the input appears to change the output distribution.

**to reproduce:**

```bash
# GPU -- broken output
llama-server -m GLM-4.7-Flash-Q4_K_M.gguf --port 8012 \
  --ctx-size 4096 --n-gpu-layers 99 --flash-attn off

curl -s -X POST http://127.0.0.1:8012/v1/chat/completions \
  -H "Content-Type: application/json" \
  -d '{"messages":[{"role":"user","content":"What is 7 times 13?"}],"max_tokens":10,"temperature":0,"logprobs":true,"top_logprobs":3}'
# -> "??????????" every time, static top_logprobs

# CPU -- correct output
llama-server -m GLM-4.7-Flash-Q4_K_M.gguf --port 8012 \
  --ctx-size 4096 --n-gpu-layers 0

# same curl -> "91"
```

things tried: 
- flash attention: tested with `--flash-attn off`, same result
- gating function: startup confirms `expert_gating_func = sigmoid` (correct for GLM)
- sampling: tested temp 0, min-p 0.01, repeat-penalty 1.0
- model file: three independent GGUF sources all reproduce
- GPU hardware: Qwen3-30B-A3B (standard MoE, not deepseek2) runs perfectly at 152 t/s on the same card
- quantization: different quants produce different garbage tokens but the same static-logit pattern

the model generates tokens at ~96 t/s on GPU (fast, but wrong). prompt eval looks normal. the compute graph runs, it just produces constant logits.

different GGUF providers produce different degenerate tokens (bartowski: `?`, unsloth: `{n;;`, ollama blob: multilingual gibberish)

### First Bad Commit

haven't been able to bisect this fully. tested b8323 and b8363, both broken. the model might have never worked correctly on CUDA Ampere via llama-server -- I was previously running it through Ollama which may have been silently falling back to CPU.

### Relevant log output

<details>
<summary>Logs</summary>

```console
$ llama-server -m GLM-4.7-Flash-Q4_K_M.gguf --port 8012 --ctx-size 4096 --n-gpu-layers 99 --flash-attn off --reasoning off

general.architecture = deepseek2
n_expert = 64, n_expert_used = 4
expert_gating_func = sigmoid
expert_weights_scale = 1.8
expert_weights_norm = 1
Flash Attention = disabled

load_tensors: offloading output layer to GPU
load_tensors: offloaded 49/49 layers to GPU
load_tensors: CUDA0 model buffer size = 17680.81 MiB

print_info: arch = deepseek2
print_info: n_vocab = 154880
print_info: BOS token = 154822
print_info: EOS token = 154820 '<|endoftext|>'

common_init_result: added <|endoftext|> logit bias = -inf
common_init_result: added <|user|> logit bias = -inf
common_init_result: added <|observation|> logit bias = -inf
```

</details>
